cumulus/polkadot-parachain/src/service.rs

```diff
@@ -526,403 +557,379 @@ where
-	sync_oracle: Arc<SyncingService<Block>>,
-	keystore: KeystorePtr,
-	relay_chain_slot_duration: Duration,
-	para_id: ParaId,
-	collator_key: CollatorPair,
-	overseer_handle: OverseerHandle,
-	announce_block: Arc<dyn Fn(Hash, Option<Vec<u8>>) + Send + Sync>,
-	backend: Arc<ParachainBackend>,
-) -> Result<(), sc_service::Error>
+pub struct StartLookaheadAuraConsensus<RuntimeApi>(PhantomData<RuntimeApi>);
+
+impl<RuntimeApi> StartConsensus<RuntimeApi> for StartLookaheadAuraConsensus<RuntimeApi>
 where
 	RuntimeApi: ConstructNodeRuntimeApi<Block, ParachainClient<RuntimeApi>>,
 	RuntimeApi::RuntimeApi: AuraRuntimeApi<Block, AuraId>,
 {
-	let info = backend.blockchain().info();
-	if !client.runtime_api().has_aura_api(info.finalized_hash) {
-		return Err(sc_service::error::Error::Other("Missing aura runtime APIs".to_string()));
+	fn start_consensus(
+		client: Arc<ParachainClient<RuntimeApi>>,
+		block_import: ParachainBlockImport<RuntimeApi>,
+		prometheus_registry: Option<&Registry>,
+		telemetry: Option<TelemetryHandle>,
+		task_manager: &TaskManager,
+		relay_chain_interface: Arc<dyn RelayChainInterface>,
+		transaction_pool: Arc<FullPool<Block, ParachainClient<RuntimeApi>>>,
+		sync_oracle: Arc<SyncingService<Block>>,
+		keystore: KeystorePtr,
+		relay_chain_slot_duration: Duration,
+		para_id: ParaId,
+		collator_key: CollatorPair,
+		overseer_handle: OverseerHandle,
+		announce_block: Arc<dyn Fn(Hash, Option<Vec<u8>>) + Send + Sync>,
+		backend: Arc<ParachainBackend>,
+	) -> Result<(), Error> {
+		let info = backend.blockchain().info();
+		if !client.runtime_api().has_aura_api(info.finalized_hash) {
+			return Err(sc_service::error::Error::Other("Missing aura runtime APIs".to_string()));
+		}
+
+		let proposer_factory = sc_basic_authorship::ProposerFactory::with_proof_recording(
+			task_manager.spawn_handle(),
+			client.clone(),
+			transaction_pool,
+			prometheus_registry,
+			telemetry.clone(),
+		);
+
+		let collator_service = CollatorService::new(
+			client.clone(),
+			Arc::new(task_manager.spawn_handle()),
+			announce_block,
+			client.clone(),
+		);
+
+		let params = AuraParams {
+			create_inherent_data_providers: move |_, ()| async move { Ok(()) },
+			block_import,
+			para_client: client.clone(),
+			para_backend: backend,
+			relay_client: relay_chain_interface,
+			code_hash_provider: move |block_hash| {
+				client.code_at(block_hash).ok().map(|c| ValidationCode::from(c).hash())
+			},
+			sync_oracle,
+			keystore,
+			collator_key,
+			para_id,
+			overseer_handle,
+			relay_chain_slot_duration,
+			proposer: Proposer::new(proposer_factory),
+			collator_service,
+			authoring_duration: Duration::from_millis(1500),
+			reinitialize: false,
+		};
+
+		let fut =
+			aura::run::<Block, <AuraId as AppCrypto>::Pair, _, _, _, _, _, _, _, _, _>(params);
+		task_manager.spawn_essential_handle().spawn("aura", None, fut);
+
+		Ok(())
 	}
+}
+
+pub struct StartRelayToAuraConsensus<RuntimeApi, AuraId>(PhantomData<(RuntimeApi, AuraId)>);
```

Comment:
```suggestion
pub(crate) struct StartRelayToAuraConsensus<RuntimeApi, AuraId>(PhantomData<(RuntimeApi, AuraId)>);
```

Here and everywhere else should be enough

Reply:
Done